vscode-dotnet-runtime-library/src/Acquisition/WinMacGlobalInstaller.ts

```diff
@@ -0,0 +1,308 @@
+/* --------------------------------------------------------------------------------------------
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ * ------------------------------------------------------------------------------------------ */
+import * as fs from 'fs';
+import * as os from 'os';
+import * as path from 'path';
+import * as https from 'https';
+
+import { FileUtilities } from '../Utils/FileUtilities';
+import { IGlobalInstaller } from './IGlobalInstaller';
+import { IAcquisitionWorkerContext } from './IAcquisitionWorkerContext';
+import { VersionResolver } from './VersionResolver';
+import { DotnetConflictingGlobalWindowsInstallError, DotnetCustomLinuxInstallExistsError, DotnetUnexpectedInstallerOSError } from '../EventStream/EventStreamEvents';
+import { ICommandExecutor } from '../Utils/ICommandExecutor';
+import { CommandExecutor } from '../Utils/CommandExecutor';
+/* tslint:disable:only-arrow-functions */
+/* tslint:disable:no-empty */
+
+/**
+ * @remarks
+ * This class manages global .NET SDK installations for windows and mac.
+ * Both of these OS's have official installers that we can download and run on the machine.
+ * Since Linux does not, it is delegated into its own set of classes.
+ */
+export class WinMacGlobalInstaller extends IGlobalInstaller {
+
+    private installerUrl : string;
+    private installingVersion : string;
+    protected commandRunner : ICommandExecutor;
+    public cleanupInstallFiles = true;
+
+    constructor(context : IAcquisitionWorkerContext, installingVersion : string, installerUrl : string, executor : ICommandExecutor | null = null)
+    {
+        super(context);
+        this.installerUrl = installerUrl
+        this.installingVersion = installingVersion;
+        this.commandRunner = executor ?? new CommandExecutor();
+    }
+
+    public async installSDK(): Promise<string>
+    {
+        // Check for conflicting windows installs
+        if(os.platform() === 'win32')
+        {
+            const conflictingVersion = await this.GlobalWindowsInstallWithConflictingVersionAlreadyExists(this.installingVersion);
+            if(conflictingVersion !== '')
+            {
+                if(conflictingVersion === this.installingVersion)
+                {
+                    // The install already exists, we can just exit with Ok.
+                    return '0';
+                }
+                const err = new DotnetConflictingGlobalWindowsInstallError(new Error(`An global install is already on the machine: version ${conflictingVersion}, that conflicts with the requested version.
```

Comment:
On Windows, are you basing the detection off the registry for the SDK or runtime? If it was acquired through VS, users won't be able to remove it because of reference counts and also because VS will hide the installations in ARP so this could be telling users to do something they cannot do.

Reply:
I'm basing it off the registry keys of the SDK. The code to check the registry is in this file in the function `getGlobalSdkVersionsInstalledOnMachine`. That's a good point, in this case: could VS completely hide the fact that this SDK was installed in the setup component dialogue, or could the users uninstall it there?

@baronfel Should we remediate this error message to suggest that users may need to uninstall components in VS as well?

Reply:
Yes, we should try to align the messages with what we did in the SDK itself for detecting these conflicts if at all possible.

Reply:
I added some more details to the error message about how to uninstall via VS Setup